crates/matrix-sdk/src/client/mod.rs

```diff
@@ -952,6 +952,48 @@ impl Client {
         self.login_with_token(token.as_str(), device_id, initial_device_display_name).await
     }
 
+    /// Login as appservice virtual user
+    ///
+    /// This function can be used to create an end-to-end-encryption capable
+    /// appservice virtual user.
+    ///
+    /// # Arguments
+    ///
+    /// * `user` - The user that should be logged in to the homeserver.
+    ///
+    /// * `device_id` - A unique id that will be associated with this session.
+    ///   If not given the homeserver will create one. Can be an existing
+    ///   device_id from a previous login call. Note that this should be done
+    ///   only if the client also holds the encryption keys for this device.
+    #[instrument(skip(self, user))]
+    #[cfg(feature = "appservice")]
+    pub async fn appservice_session(
+        &self,
+        user: impl AsRef<str>,
+        device_id: Option<&str>,
+        initial_device_display_name: Option<&str>,
+    ) -> Result<Session> {
+        let homeserver = self.homeserver().await;
+        info!(homeserver = homeserver.as_str(), user = user.as_ref(), "Logging in");
```

Comment:
Could we distinguish this log line so it's clearer that it's not a normal login flow.